src/main/java/yablunovskyi/bookstore/mapper/BookMapper.java

```diff
@@ -2,21 +2,31 @@
 
 import org.mapstruct.InjectionStrategy;
 import org.mapstruct.Mapper;
+import org.mapstruct.Mapping;
 import org.mapstruct.MappingTarget;
 import org.mapstruct.NullValueCheckStrategy;
-import yablunovskyi.bookstore.dto.book.BookDto;
-import yablunovskyi.bookstore.dto.book.CreateBookRequestDto;
+import yablunovskyi.bookstore.dto.book.BookRequestDto;
+import yablunovskyi.bookstore.dto.book.BookResponseDto;
+import yablunovskyi.bookstore.dto.book.BookResponseDtoWithoutCategoryIds;
 import yablunovskyi.bookstore.model.Book;
+import yablunovskyi.bookstore.model.Category;
 
 @Mapper(
         componentModel = "spring",
         injectionStrategy = InjectionStrategy.CONSTRUCTOR,
         nullValueCheckStrategy = NullValueCheckStrategy.ALWAYS
 )
 public interface BookMapper {
-    BookDto toDto(Book book);
+    @Mapping(source = "categories", target = "categoriesId")
+    BookResponseDto toDto(Book book);
     
-    Book toBook(CreateBookRequestDto requestDto);
+    Book toBook(BookRequestDto requestDto);
     
-    void updateRequestDtoToBook(CreateBookRequestDto requestDto, @MappingTarget Book book);
+    void updateRequestDtoToBook(BookRequestDto requestDto, @MappingTarget Book book);
```

Comment:
Suggest adding BeanMapping annotation here specifying the nullValuePropertyMappingStrategy property. It will allow you to update only specific fields for book.